tests/conftest.py

```diff
@@ -429,6 +429,112 @@ def finalizer():
     return factory
 
 
+@pytest.fixture()
+def service_account_factory(request):
+    """
+    Create a service account
+    """
+    instances = []
+    active_service_account_obj = None
+
+    def factory(project=None):
+        """
+        Args:
+            project (object): ocs_ci.ocs.resources.ocs.OCS instance
+                of 'Project' kind.
+
+        Returns:
+            object: helpers.create_serviceaccount instance.
+        """
+        nonlocal active_service_account_obj
+        if active_service_account_obj:
+            return active_service_account_obj
+        else:
+            sa_obj = helpers.create_serviceaccount(
+                namespace=project.namespace,
+            )
+            sa_obj.project = project
+            active_service_account_obj = sa_obj
+            helpers.add_scc_policy(sa_name=sa_obj.name, namespace=project.namespace)
+            assert sa_obj, "Failed to create serviceaccount"
+            instances.append(sa_obj)
+            return sa_obj
+
+    def finalizer():
+        """
+        Delete the service account
+        """
+        for instance in instances:
+            helpers.remove_scc_policy(
+                sa_name=instance.name,
+                namespace=instance.namespace
+            )
+            instance.delete()
+            instance.ocp.wait_for_delete(resource_name=instance.name)
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
+@pytest.fixture()
+def dc_pod_factory(
+    request,
+    service_account_factory,
+    pvc_factory,
+):
+    """
+    Create deploymentconfig pods
+    """
+    instances = []
+
+    def factory(
+        interface=constants.CEPHBLOCKPOOL,
+        pvc=None,
+        size=None,
+        custom_data=None,
+        replica_count=1,
+    ):
+        """
+        Args:
+            interface (str): CephBlockPool or CephFileSystem. This decides
+                whether a RBD based or CephFS resource is created.
+                RBD is default.
+            pvc (PVC object): ocs_ci.ocs.resources.pvc.PVC instance kind.
+            size (int): The requested size for the PVC
+            custom_data (dict): If provided then Pod object is created
+                by using these data. Parameter `pvc` is not used but reference
+                is set if provided.
+            replica_count (int): Replica count for deployment config
+        """
+        if custom_data:
+            dc_pod_obj = helpers.create_resource(**custom_data)
+        else:
+
+            pvc = pvc or pvc_factory(interface=interface, size=size)
+            sa_obj = service_account_factory(project=pvc.project)
```

Comment:
User should have option to add service_account via parameter (as with pvc)

Reply:
@fbalak sorry i didn't get you

Reply:
I am sorry that I did not make myself clear. I meant to add `service_account` to factory function like:
```
   def factory(
        interface=constants.CEPHBLOCKPOOL,
        pvc=None,
        service_account=None,
        size=None,
        custom_data=None,
        replica_count=1,
    ):
```
and then fill this variable like:
```
sa_obj = service_account or service_account_factory(project=pvc.project)
```
This would allow user that uses `dc_pod_factory` to control what service account is used in created pod object.

Reply:
@fbalak service_account should be created under newly created namespace be cannot add service account from other namespace dc pod won't come in running state

Reply:
Right, but if you add the parameter then you will have option to test adding service_account from different namespace it as negative testcase or create two pods with different user accounts in same namespace.

Reply:
Done